src/anyio/_backends/_asyncio.py

```diff
@@ -1718,8 +1722,8 @@ async def send(self, item: bytes) -> None:
                     return
 
 
-_read_events: RunVar[dict[Any, asyncio.Event]] = RunVar("read_events")
-_write_events: RunVar[dict[Any, asyncio.Event]] = RunVar("write_events")
+_read_events: RunVar[dict[socket.socket | int, asyncio.Event]] = RunVar("read_events")
```

Comment:
```suggestion
_read_events: RunVar[dict[int, asyncio.Event]] = RunVar("read_events")
```

Reply:
It shouldn't be possible to put an int and a socket in here, otherwise a socket with the same fileno as an int could be added